provider/src/commonMain/kotlin/at/asitplus/crypto/provider/sign/SignatureInput.kt

```diff
@@ -0,0 +1,62 @@
+package at.asitplus.crypto.provider.sign
+
+import at.asitplus.catching
+import at.asitplus.crypto.datatypes.Digest
+import at.asitplus.crypto.datatypes.asn1.ensureSize
+import at.asitplus.crypto.datatypes.misc.BitLength
+import at.asitplus.crypto.provider.hash.digest
+import com.ionspin.kotlin.bignum.integer.BigInteger
+import com.ionspin.kotlin.bignum.integer.Sign
+
+typealias SignatureInputFormat = Digest?
+private val RAW_BYTES: SignatureInputFormat = null
+class SignatureInput private constructor (
+    val data: Sequence<ByteArray>,
+    val format: SignatureInputFormat
+){
+
+    companion object {
+        /** only use this if you know what you are doing */
+        fun unsafeCreate(data: ByteArray, format: SignatureInputFormat): SignatureInput {
+            if (format != null)
+                require(data.size == format.outputLength.bytes.toInt())
+
+            return SignatureInput(sequenceOf(data), format)
+        }
+    }
+
+    fun convertTo(format: SignatureInputFormat) = catching {
+        if (this.format == format) return@catching this
+        if (this.format != RAW_BYTES) throw IllegalStateException("Cannot convert from ${this.format} to $format")
+        format!! /* RAW_BYTES is null; this is for the compiler */
+        SignatureInput(sequenceOf(format.digest(this.data)), format)
+    }
+
+    constructor(data: ByteArray) : this(sequenceOf(data), RAW_BYTES)
+    constructor(data: Sequence<ByteArray>): this(data, RAW_BYTES)
+
+    /**
+     * Takes the leftmost bits of the byte array, and converts them to an unsigned `BigInteger`.
+     *
+     * (This matches the ECDSA spec.)
+     */
+    fun asBigInteger(length: BitLength): BigInteger {
```

Comment:
KmmResult?

Reply:
there's no useful error conditions to catch

Reply:
Soooo it's an IO error?!